nexus-sdk/src/main/java/io/nexusrpc/handler/OperationContext.java

```diff
@@ -68,21 +75,51 @@ public boolean isMethodCancelled() {
    * Add a listener for method cancellation. This will be invoked immediately before this function
    * returns if the method is already cancelled. The listener must not block. This is not reentrant
    * and therefore must not be called in another cancellation listener.
+   *
+   * @return this
    */
-  public void addMethodCancellationListener(OperationMethodCancellationListener listener) {
+  public OperationContext addMethodCancellationListener(
+      OperationMethodCancellationListener listener) {
     if (methodCanceller != null) {
       methodCanceller.addListener(listener);
     }
+    return this;
```

Comment:
Meh, not sure the context needs to return itself in a builder-pattern style IMO

Reply:
This is how similar API like otel Spans work

Reply:
Sure, lots of libraries `return this` when adding things. And lots of libraries don't. Taken to its logical conclusion, nobody would ever use `void` and point to libraries like OTel for justification. This will be even more confusing if there's ever a copy-on-write function that returns a new instance of the context and we have to make sure developers closely read the javadoc. I do not think it's a good pattern to return this instead of void for methods like these outside of a clear builder situation.

Having said that, I won't make this blocking.